multicast/recv.py

```diff
@@ -104,34 +89,172 @@
+		True
+		>>>
+
+	Testcase 0: Recv should be automaticly imported.
+		A: Test that the multicast component is initialized.
+		B: Test that the recv component is initialized.
+
+		>>> multicast is not None
+		True
+		>>> multicast.recv is not None
+		True
+		>>>
+
+	Testcase 1: Recv should have genSocket() function that returns a socket.socket object.
+		A: Test that the recv component has the function 'genSocket'
+		B: Test that the 'genSocket' function returns a socket
+
+		>>> multicast.recv.genSocket is not None
+		True
+		>>> multicast.recv.genSocket #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
+		<function genSocket at ...>
+		>>> type(multicast.recv.genSocket)
+		<class 'function'>
+		>>> type(multicast.recv.genSocket())
+		<class 'socket.socket'>
+		>>>
+
+
+	"""
+	sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+	# allow reuse of socket (to allow another instance of python running this
+	# script binding to the same ip/port)
+	sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+	sock.settimeout(20)
+	return sock
+
+
+def endSocket(sock=None):
+	"""Will generates an unbound socket.socket object ready to receive network traffic.
+
+	Minimal Acceptance Testing:
+
+	First setup test fixtures by importing multicast.
+
+		>>> import multicast
+		>>> multicast.__doc__ is not None
+		True
+		>>>
+
+	Testcase 0: Recv should be automaticly imported.
+		A: Test that the multicast component is initialized.
+		B: Test that the recv component is initialized.
+
+		>>> multicast is not None
+		True
+		>>> multicast.recv is not None
+		True
+		>>>
+
+	Testcase 1: Recv should have endSocket() function that takes a socket.socket and closes it.
+		A: Test that the recv component has the function 'genSocket'
+		B: Test that the recv component has the function 'endSocket'
+		C: Test that the 'endSocket' function returns None when given the genSocket
+
+		>>> multicast.recv.genSocket is not None
+		True
+		>>> multicast.recv.endSocket is not None
+		True
+		>>> multicast.recv.endSocket #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
+		<function endSocket at ...>
+		>>> type(multicast.recv.endSocket)
+		<class 'function'>
+		>>> temp_fxtr = multicast.recv.endSocket(multicast.recv.genSocket())
+		>>> temp_fxtr is None
+		True
+		>>>
+
+	Testcase 2: Recv should have endSocket() function that takes a socket.socket object,
+		otherwise does nothing.
+		A: Test that the recv component has the function 'endSocket' (see testcase 1)
+		B: Test that the 'endSocket' function returns nothing
+
+		>>> multicast.recv.endSocket is not None
+		True
+		>>> multicast.recv.endSocket #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
+		<function endSocket at ...>
+		>>> type(multicast.recv.endSocket)
+		<class 'function'>
+		>>> multicast.recv.endSocket(None) is None
+		True
+		>>>
+
+
+	"""
+	if not (sock is None):  # pragma: no branch
+		try:
+			sock.close()
+			sock.shutdown(socket.SHUT_RD)
+		except OSError:
+			sock = None
+
```

Comment:
🤔 Does this leak closed `sock` objects?

- [ ] open issue to explore leaks in python nuances
- [ ] resolve this convo for this PR

Reply:
see #35